### PR TITLE
Added new file for yagni principle

### LINKED ISSUE #318: New article: YAGNI principle
# Title: YAGNI principle

- [ ] What is this about: “What is this practice about”? Which existing (or new) problems does this practice solve? 
Don’t codify more than you’ve been asked to. Don’t anticipate the future, and just create something that works as soon as possible. Code only the necessary parts to solve the current problem.

- [ ] Read more here: The starting point for your research (You can use different sources if you have better ones)
-https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it

- [ ] Pick the corresponding labels: 
    - Stage:research
    - Stage:building_team
    - **Stage:development**
    - Stage:launch
    - Stage:maintenance
    - Stage:phase_out

## How to contribute:
- Follow our [Contribution guide](https://github.com/DXHeroes/knowledge-base-content/blob/master/CONTRIBUTING.md)
- How to [Git and Pull Requests]( https://github.com/firstcontributions/first-contributions/blob/master/README.md)
- How to [Markdown](https://guides.github.com/features/mastering-markdown/), [Markdown interactive demo](https://www.markdowntutorial.com/lesson/1/) or [Google Docs Addon](https://gsuite.google.com/marketplace/app/docs_to_markdown/700168918607)

### PR DESCRIPTION
**Pull request by ShvetaVasisht**
Closes #318 

## If an article related PR

- [X] I have checked that my suggestion follows the Contribution guide

- [X]  I have added which issue this PR closes

- [ ] I have 0 warnings from Danger.js

- [X] If I have added images, they are in /files/ folder

## If other PR

- [ ] I have added which issue this PR closes

### CODE REVIEW

**Review by prokopsimek — APPROVED** [review]
@ShvetaVasisht Hello! The pull request looks great! 👏  Can I ask you how much time it took to write your article? Just to collect some stats because the article has the all necessary information and is well short. 😍

@ridlees May I ask you for the second review? Thanks!

**Comment by ShvetaVasisht** [comment]
Thank you @prokopsimek .I think it took around 6 to 8 hours.

File: practices/yagni_principle.md
Comment (ridlees) [inline practices/yagni_principle.md:72]:
```suggestion
Ask yourself whether you will need these features  soon or if you are just assuming they might be important.```((

Suggestion for clarity's sake, to show that the important distinction is in "will you need them or are you assuming that you will need them". In the original, a non-native speaker can instead think that the distinction is in the timing (soon vs distant future).

File: practices/yagni_principle.md
Comment (ridlees) [inline practices/yagni_principle.md:41]:
```suggestion
YAGNI is an extreme programming principle (XP) which states that you must not implement features that you assume as significant in the future. 
```
Consider this suggestion, don't merge if it doesn't feel right.

File: practices/yagni_principle.md
Comment (ridlees) [inline practices/yagni_principle.md:50]:
```suggestion
This principle helps you to focus on the required features and not add unnecessary functionality. As a programmer, you sometimes try to foresee the future. You presume that additional requirements might come up later in development. Then, there is an assumption that the present code should encapsulate this. 
```

Comment (ridlees) [reply practices/yagni_principle.md:50]: Something I would consider, it's up to you if you want to merge it :)

File: practices/yagni_principle.md
Comment (ridlees) [inline practices/yagni_principle.md:68]:
```suggestion
You need to have a clear picture of the requirements and targets to prevent deviations from the required design.```

**Review by ridlees — APPROVED** [review]
Hello @ShvetaVasisht! thanks for your great article, it's short, well-written, and sweet, and  I like it! Added some small suggestions but they are only a "consideration" (I am approving the merge request right now, so its up to you if you find the helpful).

@prokopsimek Thanks for tagging! The notification system on my end is a mess.

**Comment by prokopsimek** [comment]
@ShvetaVasisht Please, consider the suggested changes and let me know that the PR is ready to merge. 👏

**Comment by ShvetaVasisht** [comment]
@ridlees thank you for the suggestions. @prokopsimek I have applied the changes.

**Comment by ridlees** [comment]
@ShvetaVasisht @prokopsimek Then, I am merging this PR. Good job, everyone 🎉 🎉 🎉

**Pull request merged**
